xrpl/model/transactions/validations_xrpl_objects.go

```diff
@@ -88,11 +74,19 @@ func IsAmount(amount interface{}) bool {
 }
 
 // IsIssuedCurrency checks if the given object is a valid IssuedCurrency object.
-func IsIssuedCurrency(input map[string]interface{}) bool {
-	return len(maputils.GetKeys(input)) == ISSUED_CURRENCY_SIZE &&
-		typecheck.IsString(input["value"]) &&
-		typecheck.IsString(input["issuer"]) &&
-		typecheck.IsString(input["currency"])
+func IsIssuedCurrency(input interface{}) bool {
+	// Check if the input is a map.
+	if !typecheck.IsMap(input) {
+		return false
+	}
+
+	// Type the input as a map.
+	i := input.(map[string]interface{})
+
+	return len(maputils.GetKeys(i)) == ISSUED_CURRENCY_SIZE &&
+		typecheck.IsString(i["value"]) &&
```

Comment:
We can replace these checks by using:
```go
value, ok := i["value"].(string)
```

Reply:
This is what `IsString` is doing under the hood.
It seems to me it is more readable the way it is now, no ?